src/Grpc.Net.Client/GrpcChannel.cs

```diff
@@ -90,21 +95,29 @@ internal GrpcChannel(Uri address, GrpcChannelOptions channelOptions) : base(addr
             }
         }
 
-        private static HttpClient CreateInternalHttpClient()
+        private static HttpMessageInvoker CreateInternalHttpInvoker(HttpMessageHandler? handler)
         {
-            var httpClient = new HttpClient();
-
-            // Long running server and duplex streaming gRPC requests may not
-            // return any messages for over 100 seconds, triggering a cancellation
-            // of HttpClient.SendAsync. Disable timeout in internally created
-            // HttpClient for channel.
-            //
-            // gRPC deadline should be the recommended way to timeout gRPC calls.
-            //
-            // https://github.com/dotnet/corefx/issues/41650
-            httpClient.Timeout = Timeout.InfiniteTimeSpan;
-
-            return httpClient;
+            // HttpMessageInvoker should always dispose handler if Disposed is called on it.
+            // Decision to dispose invoker is controlled by _shouldDisposeHttpClient.
+            var httpInvoker = new HttpMessageInvoker(handler ?? new HttpClientHandler(), disposeHandler: true);
```

Comment:
What are the expectations around proxies with the library?  If the expectation is that this will pick up the system's proxy settings, then this is fine.  If that's not needed, there's potentially some additional perf to be had by setting UseProxy = false on the handler.

Same question about redirects.  Is the expectation that HTTP redirects will automatically be followed?  If no, you can reduce some overhead by setting AllowAutoRedirect to false.

Reply:
I think both of those are wanted. Sticking with defaults will avoid surprise, and there is always the option to turn them off if you want.

Do they have much performance impact?

Reply:
UseProxy can, in particular on Windows; depends on Windows settings.

AllowAutoRedirect has a small per-request overhead, but probably not noticeable for anything other than a memory trace.